### ISSUE TITLE
Choice whether to allow or not download in shared albums

### ISSUE THREAD

**Issue by clamyahoo** [pulsejet/memories#1027]
**Is your feature request related to a problem? Please describe.**
When I share an album with a public link, I don't always want people to be able to download the photos. But if somebody opens the link, there is the download button displayed with the photos. 

**Describe the solution you'd like**
In the Photos App, the download button isn't displayed. So somehow it should be possible to hide it and the best thing would be if the user who shares the album had the choice to hide the button just like he can with normal files in Nextcloud.
And - for the records: In the Photos App, the receiver of shared photos can download single photos. This should be regularized as well. 

**Describe alternatives you've considered**
Actually, if I want less photos to be downloaded, I share an album of Photos App. If I want no photos to be downloaded, I copy the whole album in a folder and share it blocking the download permission. But all this together is a bit weird. When it is finished, it should be possible in one place, shouldn't it?

Thank you very much and kind regards

**Comment by Miswu13** [comment]
I see this as a bug, not as a feature request. I can confirm that even if in the sharing options of the album, the download option is unticked, in the memories app the download button is still present. I don't think that this should be the normal use case.

**Comment by pulsejet** [comment]
Where are these sharing options you're referring to?

**Comment by Miswu13** [comment]
Nextcloud - > Files - > Sharing - > add user to share participants - > Custom permisions for that user - > Checkbox Allow download (see screenshot). 
I'm using nextcloud Aio v8.0.0, with
Nextcloud Hub 7 (28.0.3), and Memories 7.0.2. Android Memories app version is 1.7 
![20240325_082616](https://github.com/pulsejet/memories/assets/58112389/2fc6fc75-0ee0-4ba1-836a-e1916918da02)

**Comment by pulsejet** [comment]
@Miswu13 this is unrelated. You are referring to _folder_ shares, where there might indeed be a bug (shared files aren't treated as special, so this could be non-trivial to fix; besides it only "hides" the download button but the user can still download).

This issue is about _album_ shares, particularly public links. This is completely different from folders.

**Comment by Miswu13** [comment]
Sorry, my bad. I misunderstood the topic. For public links, I already have disabled the buttons, as well as disabling right click. My only problem was the application download button. Hopefully it will be fixed in the next release. Thank you for your time and patience.

**Comment by nkinkade** [comment]
I agree that it would be nice to be able remove/hide the big "Download" button on shared albums. Some albums may potentially have dozens of gigabytes of photos (or more), and generally speaking nobody needs to download thousands of photos at once from a large album. I can see the utility of the feature for smaller albums, so, if possible, ideally it would be a per-album option whether to display the button or not.

I understand that even if the link is removed/hidden it may still be possible for someone to manually craft the download URL, as this may be some default feature of how the Photos app has implemented albums. That said, I think it would be sufficient to simply not display the button, since only more sophisticated users will know that this is even a possibility.

**Comment by arnowelzel** [comment]
As a workaround you can use [Custom CSS](https://apps.nextcloud.com/apps/theming_customcss) and the following rule to hide the download button for shared albums:

```
/* Hide "Download" button for shared "Memories" albums */

a[href*="/apps/memories/"] {
  display: none !important;
}
```

Or to make sure, you really hide the correct links only (full path from the beginning) - replace `your-nextcloud.example` with your own domain name for Nextcloud:

```
/* Hide "Download" button for shared "Memories" albums */

a[href^="https://your-nextcloud.example/apps/memories/"] {
  display: none !important;
}
```

**Comment by clamyahoo** [comment]
Thank you @arnowelzel, this works so far on the desktop and the android app. And good to know about the Custom CSS app.

Still not perfect because of the possibility of downloading single photos in the slide show mode, but it does make a difference.

**Comment by arnowelzel** [comment]
Single Images can always be downloaded using the context menu of the browser - right click the image and choose "Save image" or a similar function. Literally everything a browser displays can be downloaded since the data was fetched from the server anyway.
